Elf/__init__.py

```diff
@@ -517,10 +517,14 @@ def getNotes(self):
                 continue
 
             notebytes =  self.readAtOffset(sec.sh_offset, sec.sh_size)
-
             offset = 0
-            while offset < len(notebytes):
+            notebyteslen = len(notebytes)
+            while offset < notebyteslen:
                 note = vs_elf.ElfNote()
+                if notebyteslen - offset < len(note):
+                    print ("\nNOTES section length mismatch!\n\t%s\n\tSection Bytes: %s\n\tStranded bytes: %s\n" % (sec, repr(notebytes), repr(notebytes[offset:])))
```

Comment:
i'd love to comply, but i strongly oppose "silent magic".  is there a better way to log this that is in line with the direction of this family of tools?  because this is in Elf/ it's not necessarily going to have a workspace to log to.  if i raise an exception, analysis fails.  would you accept the logging option williballenthin suggests above?